File: tests/cross_driver/java/src/main/java/com/secantus/smokes/ScramSha1Smoke.java
package com.secantus.smokes;

import com.mongodb.ConnectionString;
import com.mongodb.MongoClientSettings;
import com.mongodb.MongoCredential;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import org.bson.Document;

import java.util.List;

/**
 * Cross-driver SCRAM-SHA-1 smoke — Java.
 * <p>
 * Provisions a user with {@code mechanisms: [SCRAM-SHA-1]}, connects
 * with the explicit SCRAM-SHA-1 credential, and verifies
 * connectionStatus surfaces the authenticated user. Exercises the
 * legacy MD5-prepass + SHA-1 PBKDF2 path.
 */
public final class ScramSha1Smoke {

    private ScramSha1Smoke() {}

    public static void main(String[] args) {
        String uri = System.getenv("MONGODB_URI");
        String adminPwd = System.getenv("ADMIN_PASSWORD");
        if (uri == null || adminPwd == null) {
            System.err.println("MONGODB_URI and ADMIN_PASSWORD required");
            System.exit(2);
        }

        try (MongoClient root = MongoClients.create(buildSettings(uri,
                MongoCredential.createScramSha256Credential("root", "admin", adminPwd.toCharArray())))) {
            root.getDatabase("admin").runCommand(new Document()
                .append("createUser", "legacy_java")
                .append("pwd", "pass")
                .append("roles", List.of())
                .append("mechanisms", List.of("SCRAM-SHA-1")));
        }

        try (MongoClient cli = MongoClients.create(buildSettings(uri,
                MongoCredential.createScramSha1Credential("legacy_java", "admin", "pass".toCharArray())))) {
            Document status = cli.getDatabase("admin").runCommand(new Document("connectionStatus", 1));
            Document authInfo = status.get("authInfo", Document.class);
            if (authInfo == null) {
                System.err.println("connectionStatus: no authInfo");
                System.exit(1);
            }
            @SuppressWarnings("unchecked")
            List<Document> users = (List<Document>) authInfo.get("authenticatedUsers");
            if (users == null || users.stream().noneMatch(u -> "legacy_java".equals(u.getString("user")))) {
                System.err.printf("authenticatedUsers: %s%n", users);
                System.exit(1);
            }
            System.out.println("OK");
        }
    }

    private static MongoClientSettings buildSettings(String uri, MongoCredential credential) {
        return MongoClientSettings.builder()
            .applyConnectionString(new ConnectionString(uri))
            .credential(credential)
            .applyToClusterSettings(b -> b.serverSelectionTimeout(5,
                java.util.concurrent.TimeUnit.SECONDS))
            .build();
    }
}

File: tests/cross_driver/node/scram_sha1_smoke.js
// Cross-driver SCRAM-SHA-1 smoke — Node.

const { MongoClient } = require('mongodb');

function fail(msg) {
  console.error(msg);
  process.exit(1);
}

function clientFor(uri, user, pwd, authSource, mech) {
  return new MongoClient(uri, {
    auth: { username: user, password: pwd },
    authSource,
    authMechanism: mech,
    serverSelectionTimeoutMS: 5000,
  });
}

async function main() {
  const uri = process.env.MONGODB_URI;
  const adminPwd = process.env.ADMIN_PASSWORD;
  if (!uri || !adminPwd) fail('MONGODB_URI and ADMIN_PASSWORD required');

  const root = clientFor(uri, 'root', adminPwd, 'admin', 'SCRAM-SHA-256');
  await root.connect();
  try {
    await root.db('admin').command({
      createUser: 'legacy_node',
      pwd: 'pass',
      roles: [],
      mechanisms: ['SCRAM-SHA-1'],
    });
  } finally {
    await root.close();
  }

  const cli = clientFor(uri, 'legacy_node', 'pass', 'admin', 'SCRAM-SHA-1');
  await cli.connect();
  try {
    const status = await cli.db('admin').command({ connectionStatus: 1 });
    const authedUsers = status?.authInfo?.authenticatedUsers ?? [];
    if (
      authedUsers.length === 0 ||
      !authedUsers.find((u) => u.user === 'legacy_node')
    ) {
      fail(`authenticatedUsers: ${JSON.stringify(authedUsers)}`);
    }
    console.log('OK');
  } finally {
    await cli.close();
  }
}

main().catch((err) => fail(`uncaught: ${err && err.stack ? err.stack : err}`));
